### PR TITLE
chore: Issue 1416: Add unit tests for PublisherHandler lifecycle + LiveStreamPublisherManager.closeBlock()

### LINKED ISSUE #1416: Multi-Publisher General Cleanup
### Story Form

As a Block Node Engineer
I want to do cleanup post rework
So that I address todos and shortcomings

### Technical Notes

Note: All non-unit-test items are resolved.

#### List
```text
LiveStreamPublisherManagerTest.java:            // @todo(1416) Add tests for this method.
PublisherHandlerTest.java:            // @todo(1416) add tests for onComplete, not finished yet
PublisherHandlerTest.java:            // @todo(1416) add tests for onSubscribe, not finished yet
PublisherHandlerTest.java:            // @todo(1416) add tests for clientEndStreamReceived, not finished yet
PublisherHandlerTest.java:            // @todo(1416) add tests for sendAcknowledgement, not finished yet
```

### PR DESCRIPTION
**Pull request by rockysingh**
**What’s in this PR**

1. PublisherHandler lifecycle tests
	•	onSubscribe() — verifies it’s a no-op (no side effects, no exceptions).
	•	onComplete() — verifies the reply pipeline completes exactly once with no extra interactions.
	•	clientEndStreamReceived() — verifies HTTP end-of-stream triggers a clean completion of the reply pipeline with no extra interactions.
2. LiveStreamPublisherManager tests
	•	closeBlock() — verifies metric accounting for complete vs incomplete block closure.

Resolves #1416

### CODE REVIEW

**Comment by rockysingh** [comment]
> A lot of changes here which are already in main. Seems like rebasing problem.

Thanks for noting. Fixed this just now.

**Review by ata-nas — COMMENTED** [review]
In my opinion, it is a good idea to add javadocs to tests with the user's story perspective as to what the test is aiming to verify. This greatly increases understanding as to what we are aiming for. Moreover, we can juxtapose that to the test's implementation which will act as a validation if the test is correct. Personally, I never skip adding these javadocs and being (sometimes) overly extensive in my explanation.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java [hunk 1]
```diff
@@ -486,11 +486,279 @@ void testLatestBlockNumberDuringConstruction() {
...
+                // Wait (up to 3s) for the batches metric to increase beyond its baseline.
+                awaitBatchesIncrement(beforeBatches, 3_000L);
+
+                // Post-forwarder: both onNext() and closeBlock() may schedule; expect two batches produced.
+                assertThat(managerMetrics.blockBatchesMessaged().get()).isEqualTo(beforeBatches + 2);
+                // The in-memory messaging facility should now contain exactly two batches for this block.
+                assertThat(messagingFacility.getSentBlockItems()).hasSize(2);
+                // Each batch should contain the same number of items we enqueued.
+                assertThat(messagingFacility.getSentBlockItems().get(0).blockItems()).hasSize(items.length);
+                assertThat(messagingFacility.getSentBlockItems().get(1).blockItems()).hasSize(items.length);
```
Comment (ata-nas) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java]: Why does this produce 2 item sends? Maybe I am missing something, but we should be sending each block only once to the items queue?

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java [hunk 2]
```diff
@@ -486,11 +486,279 @@ void testLatestBlockNumberDuringConstruction() {
...
+            }
+
+            /**
+             * Verifies metric gating — batches only increment after the forwarder completes.
+             */
+            @Test
+            @DisplayName("batches increment only after forwarder completes (gating)")
+            void testBatchesIncrementOnlyAfterForwarderCompletes() throws InterruptedException {
```
Comment (ata-nas) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java]: I wonder, how does this test differ from `testCompletedBlockUpdatesMetricsAndForwards`

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java
Comment (ata-nas) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java]: We could add javadocs with descriptions on tests here. Non-blocking.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java [hunk 3]
```diff
@@ -2088,39 +2089,201 @@ void testOnErrorMetrics() {
...
+                    @Override
+                    public void request(long n) {
+                        // no-op: method intentionally left blank for testing
+                    }
+
+                    @Override
+                    public void cancel() {
+                        // no-op: method intentionally left blank for testing
+                    }
+                };
```
Comment (ata-nas) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java]: Nit (non-blocking): we could extract this to fixtures in the base module.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java [hunk 4]
```diff
@@ -2088,39 +2089,201 @@ void testOnErrorMetrics() {
...
+                assertThat(repliesPipeline.getOnErrorCalls()).isEmpty();
+                assertThat(repliesPipeline.getOnSubscriptionCalls()).isEmpty();
+                assertThat(repliesPipeline.getOnCompleteCalls().get()).isZero();
+            }
+
+            @Test
+            @DisplayName("multiple calls produce multiple ACKs in order and increment metrics per call")
+            void testSendAcknowledgementMultipleCalls() {
+                // Pre-assert: clean state
+                repliesPipeline.getOnNextCalls().clear();
```
Comment (ata-nas) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java]: Why the clear, each test should be instantiating new pipelines always?

**Review by ata-nas — COMMENTED** [review]
Great work @rockysingh!

I really like the descriptive javadocs and in-line comments which help guide the reader. I personally also always do that.

Great progress. I leave cleanup suggestions.

**Comment by ata-nas** [comment]
@rockysingh your CI is failing due to `spotless`. Please make sure to apply that before pushing. On another note, we also have a quite strict check for HTML elements usage in javadocs. It is possible that sometimes that will fail as well, but CI will let you know exactly where the problem is.

**Comment by rockysingh** [comment]
> Great work @rockysingh!
> 
> I really like the descriptive javadocs and in-line comments which help guide the reader. I personally also always do that.
> 
> Great progress. I leave cleanup suggestions.

Thank you - I appreciate the feedback. I will go through the feedback.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java [hunk 5]
```diff
@@ -490,7 +490,20 @@ void testLatestBlockNumberDuringConstruction() {
         @Nested
         @DisplayName("closeBlock() Tests")
         class CloseBlockTests {
-            // @todo(1416) Need to add tests for this method.
```
[the remark this answers is not in the document]
Comment (jsync-swirlds) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java]: Fixing forward limit in a separate PR is fine.

Thanks for adding these tests

Blocks must start with a Header, and must be the _next expected_ block in order to be accepted; that may be what's preventing the handler from accepting the block.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java [hunk 6]
```diff
@@ -2088,39 +2089,201 @@ void testOnErrorMetrics() {
...
+         * <ul>
+         *   <li>Pre-assert metrics and reply pipeline have no prior effects.</li>
+         *   <li>Invoke {@code sendAcknowledgement(blockNumber)}.</li>
+         *   <li>Assert a single ACKNOWLEDGEMENT response with the same {@code blockNumber}.</li>
+         *   <li>Assert {@code blockAcknowledgementsSent} incremented and no other interactions.</li>
+         * </ul>
          */
         @Nested
         @DisplayName("sendAcknowledgement() Tests")
         class SendAcknowledgementTests {
```
[the remark this answers is not in the document]
Comment (rockysingh) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java]: Thats a good edge case will add a test for it.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java [hunk 7]
```diff
@@ -486,11 +484,261 @@ void testLatestBlockNumberDuringConstruction() {
...
+         *   <li>No batch/count updates while the forwarder is still active.</li>
+         * </ul>
+         *
+         * <p><strong>Test flow (per scenario):</strong>
+         * <ol>
+         *   <li>Enqueue block items via {@code onNext(...)} and, when applicable, mark eligibility with {@code endThisBlock(...)}.</li>
+         *   <li>Call {@code closeBlock(n)} (possibly multiple times).</li>
+         *   <li>Assert immediate metrics (e.g., {@code blocksClosedComplete}).</li>
+         *   <li>Drain the executor and assert post-forwarder effects (metrics + messagingFacility payloads).</li>
+         * </ol>
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java]: A note on test javadoc.
- Javadoc should generally not explain what the code does or how the code runs; reading the code is more appropriate for that question.
- _Test_ Javadoc, in particular, should generally start by describing the requirement that the test verifies, explain how the test ensures the requirement is met, and any brittleness concerns.

### Example
```java
/*
 * Validate that the publisher manager correctly closes blocks.<br/>
 * This inner class verifies that blocks are closed, metrics are correctly updated
 * operation order is respected.
 * <p>
 * Specific items include
 * <ol>
 *   <li>Metrics are updated after batches are forwarded</li>
 *   <li>All pending batches are forwarded to messaging</li>
 *   <li>The forwarder task is correctly started or restarted, if necessary</li>
 *   <li>Metrics are not updated in the middle of sending data to messaging</li>
 * </ol>
 */
```

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java [hunk 8]
```diff
@@ -2088,39 +2089,259 @@ void testOnErrorMetrics() {
...
+             * <ul>
+             *   <li>Pre-assert: reply pipeline has no prior responses; capture current ack metric.</li>
+             *   <li>Invoke {@code sendAcknowledgement(blockNumber)} twice for the SAME block.</li>
+             *   <li>Assert: two ACK responses are emitted in order and both carry the same {@code blockNumber}.</li>
+             *   <li>Assert: {@code blockAcknowledgementsSent} increments by 2; no other side effects.</li>
+             * </ul>
+             * Rationale: guards against regressions where duplicate ACKs for the same block are coalesced or rejected.
+             */
+            @Test
+            @DisplayName("duplicate acknowledgements for same block produce two ACKs and increment metrics twice")
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java]: We actually shouldn't test for this.
Reasoning:
1. Acknowledgements are comprehensive. Acknowledgement for block `N` also acknowledges all blocks from `0`-`N-1`.
1. For efficiency, we do not require that acknowledgements are sent or received in any particular order.
1. For efficiency, we do not guarantee to acknowledge every block, only that we will acknowledge blocks in a timely manner.

Comment (rockysingh) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java]: Are you for me to remove this test?

**Review by jsync-swirlds — APPROVED** [review]
